### PR TITLE
Add linters.

### PR DESCRIPTION
**Pull request by ldez**
- add gometalinter
- fix golint issues
- add a Makefile
- update Travis configuration
- add `vendor` to gitignore.

The build will fail until the PR #32 is merged.

### CODE REVIEW

**Comment by ldez** [comment]
Now the build works :tada: 

I also fixed linting problem from #30.

File: secure.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ const (
 	cspNonceSize       = 16
 )
 
-// a type whose pointer is the type of field `SSLHostFunc` of `Options` struct
+// SSLHostFunc a type whose pointer is the type of field `SSLHostFunc` of `Options` struct
```
Comment (unrolled) [inline secure.go]: 👍

File: secure_test.go [hunk 2]
```diff
@@ -236,7 +236,7 @@ func TestBasicSSLWithHostFunc(t *testing.T) {
 	sslHostFunc := (func() SSLHostFunc {
 		isServerDown := false
 		return func(host string) (newHost string) {
-			if isServerDown == true {
+			if isServerDown {
```
Comment (unrolled) [inline secure_test.go]: 👍

**Comment by ldez** [comment]
@unrolled I can revert changes on names if you want, but for this I need to disable `golint` or exclude fields with some gometalinter directive (comments) in the code.

**Comment by unrolled** [comment]
Keeping the `// nolint: golint` lines it cool with me 👍

**Comment by ldez** [comment]
@unrolled pb fixed.

File: .gitignore [hunk 3]
```diff
@@ -24,3 +24,5 @@ _testmain.go
 
 *.pem
 .DS_Store
+
+vendor/
```
Comment (unrolled) [inline .gitignore]: Undo

File: .gometalinter.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (unrolled) [inline .gometalinter.json]: Remove file

File: .travis.yml [hunk 5]
```diff
@@ -6,8 +6,10 @@ go:
   - 1.9.x
   - tip
 
-install:
+before_install:
```
Comment (unrolled) [inline .travis.yml]: Undo

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.PHONY: all
```
Comment (unrolled) [inline Makefile]: Remove file

**Comment by ldez** [comment]
If I remove all files, my PR doesn't mean anything.

I don't understand why you don't want to add linter now.

But if you don't want the linter, close my PR.

**Comment by unrolled** [comment]
The other changes are good, I just don't want to integrate this repo with the linter directly. I have my editor doing the linting for me right now. I think this is valuable when starting a new project, but less so when it's already stable.

**Comment by ldez** [comment]
Integrating a linter in a CI is a classic for project management.
Automation has many advantages even if it's a new project or not.

The meaning of my PR is to add a linter, I will not remove the linter, it's a nonsense.

Feel free to close this PR, if you don't agree with this approach, it's not a problem for me.

**Comment by unrolled** [comment]
Thanks for your work on this and your feedback! I'm going to do some thinking / planning on how best to integrate linting on existing and future projects.

**Pull request closed without merging**